cli/src/utils.rs

```diff
@@ -33,6 +33,26 @@ pub struct FileOrUrl {
     pub version: Option<MetadataVersion>,
 }
 
+/// if `--path -` is provided, read bytes for metadata from stdin
+const STDIN_PATH_NAME: &str = "-";
+#[derive(Debug, Clone)]
+pub enum PathOrStdIn {
+    Path(PathBuf),
+    StdIn,
+}
+
+impl FromStr for PathOrStdIn {
+    type Err = <PathBuf as FromStr>::Err;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
I wonder whether we should trim the leading and trailing whitespaces before checking that the `--file is `-`

Reply:
and a simple test for the `from_str` impl would be nice to have.

Reply:
I think regarding trimming, `clap` already takes care of that before the args are given to us to parse, but I am not entirely sure.

Reply:
Tadeo added trimming and tests for it :)